ddclient.in

```diff
@@ -5170,6 +5185,113 @@ sub nic_dinahosting_update {
     }
 }
 
+######################################################################
+## nic_gandi_examples
+## by Jimmy Thrasibule <dev@jimmy.lt>
+######################################################################
+sub nic_gandi_examples {
+    return <<"EoEXAMPLE";
+o 'gandi'
+
+The 'gandi' protocol is used by the LiveDNS service offered by gandi.net.
+
+Description of Gandi's LiveDNS API can be found at:
+
+    https://api.gandi.net/docs/livedns/
+
+
+Available configuration variables:
+  * password: The Gandi API key. If you don’t have one yet, you can generate
+              your production API key from the API Key Page (in the Security
+              section).
+  * ttl:      The time-to-live value associated with the updated DNS record.
+  * zone:     The DNS zone to be updated.
+
+Example ${program}.conf file entries:
+  ## Single host update.
+  protocol=gandi,                           \\
+  zone=example.com,                         \\
+  password=my-gandi-api-key,                \\
+  ttl=3600                                  \\
+  host.example.com
+
+  ## Multiple host update.
+  protocol=gandi,                           \\
+  zone=example.com,                         \\
+  password=my-gandi-api-key,                \\
+  ttl=3600                                  \\
+  hosta.example.com,hostb.sub.example.com
+EoEXAMPLE
+}
+
+######################################################################
+## nic_gandi_update
+######################################################################
+sub nic_gandi_update {
+    debug("\nnic_gandi_update -------------------");
+
+    # Update each set configured host.
+    foreach my $h (@_) {
+        my $ip = delete $config{$h}{'wantip'};
+        (my $hostname = $h) =~ s/\.\Q$config{$h}{zone}\E$//;
+
+        info("%s -- Setting IP address to %s.", $h, $ip);
+        verbose("UPDATE:", "updating %s", $h);
+
+        my $headers;
+        $headers  = "Content-Type: application/json\n";
+        $headers .= "Authorization: Apikey $config{$h}{'password'}\n";
+
+        my $data = encode_json({
+            items => [{
+                rrset_type   => is_ipv6($ip) ? "AAAA" : "A",
+                rrset_ttl    => $config{$h}{'ttl'},
+                rrset_values => [$ip],
+            }]
+        });
+
+        my $url = "https://$config{$h}{'server'}/$config{$h}{'script'}/livedns/domains/$config{$h}{'zone'}/records/$hostname";
```

Comment:
What do you think about adding the record type to the end of this URL instead of `rrset_type` in the JSON? As it is now, users can't have a static IPv6 address with a dynamic IPv4 address because the IPv4 update will nuke the AAAA record.

Reply:
Well, I did not understand how that works. I though we had to declare two services (one for IPv4 and one for IPv6) specifically since the web method (which is my use case) only provide a single IP (if I'm getting this right).

How to have both IPv4 and IPv6?

Reply:
Never mind I got you're saying here.